testing/inference/inference_alg_test.py

```diff
@@ -43,3 +46,23 @@ def compute(self, F, variables):
 
         assert np.allclose(x_res.asnumpy(), x_np)
         assert np.allclose(y_res.asnumpy(), y_np)
+
+    def test_chagne_default_dtype(self):
+        from mxfusion.common import config
+        config.DEFAULT_DTYPE = 'float64'
+
+        np.random.seed(0)
+        mean_groundtruth = 3.
+        variance_groundtruth = 5.
+        N = 100
+        data = np.random.randn(N)*np.sqrt(variance_groundtruth) + mean_groundtruth
+
+        m = Model()
+        m.mu = Variable()
+        m.s = Variable(transformation=PositiveTransformation())
+        m.Y = Normal.define_variable(mean=m.mu, variance=m.s, shape=(100,))
+
+        infr = GradBasedInference(inference_algorithm=MAP(model=m, observed=[m.Y]))
+        infr.run(Y=mx.nd.array(data, dtype='float64'), learning_rate=0.1, max_iters=1)
```

Comment:
Just want to change to 2 iterations in case something was initialized differently than it gets updated.
```suggestion
        infr.run(Y=mx.nd.array(data, dtype='float64'), learning_rate=0.1, max_iters=2)
```